### ISSUE TITLE
vmagent k8s discovery needs cluster wide access to get pods within namespace

### ISSUE THREAD

**Issue by dmitry-tiger** [VictoriaMetrics/VictoriaMetrics#432]
**Describe the bug**
vmagent gets error "cannot list pods at the cluster scope
" from pods discovery when does discovery within namespace and container access is limited to namespace

**To Reproduce**
set up discovery job
```
- job_name: promxy-stats
  kubernetes_sd_configs:
  - role: pod
    namespaces:
      names:
      - promxy-namespace
relabel_configs:
- source_labels: [__meta_kubernetes_pod_label_app]
  regex: promxy
  action: keep
```

grant access only within namespace
 ```
 - kind: Role
    apiVersion: rbac.authorization.k8s.io/v1
    metadata:
      name: vmagent
      namespace: promxy-namespace
      labels:
        k8s-app: vmagent
    rules:
      - apiGroups: [""]
        resources:
          - pods
        verbs:
          - get
          - list
          - watch
```

**Expected behavior**
Requests limited to namespace according API 
https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.14/#list-pod-v1-core have to be like:

> GET /api/v1/namespaces/{namespace}/pods

But in logs I see request to listed below URL which requres cluster wide access 

> https://172.20.0.1:443/api/v1/pods?fieldSelector=metadata.namespace%3Dpromxy-namespace

**Version**
1.34.9

**Additional context**
log message:

> 2020-04-23T14:45:58.625Z	error	VictoriaMetrics/lib/promscrape/config.go:332	error when discovering kubernetes pods for `job_name` "promxy-stats": cannot obtain pods data from API server: unexpected status code returned from "https://172.20.0.1:443/api/v1/pods?fieldSelector=metadata.namespace%3Dpromxy-namespace":  403; expecting 200; response body: "{\"kind\":\"Status\",\"apiVersion\":\"v1\",\"metadata\":{},\"status\":\"Failure\",\"message\":\"pods is forbidden: User \\\"system:serviceaccount:promxy-namespace:vmagent\\\" cannot list pods at the cluster scope: no RBAC policy matched\",\"reason\":\"Forbidden\",\"details\":{\"kind\":\"pods\"},\"code\":403}\n"; skipping it

**Comment by valyala** [comment]
Confirmed the issue. I think the best solution would be to query `/api/v1/namespaces/*` for all the namespaces set in `kubernetes_sd_config->namespaces`.

**Comment by valyala** [comment]
This issue should be fixed in the commit 24461153bffb133676f92693cf5ae8645e1719ad . This commit will be included in the next release.

@dmitry-tiger , could you build `vmagent` from this commit according to [these docs](https://github.com/VictoriaMetrics/VictoriaMetrics/blob/master/app/vmagent/README.md#how-to-build-from-sources) and verify whether the issue is fixed there?

**Comment by dmitry-tiger** [comment]
@valyala Sorry but I don't have a possibility to build custom images in our environment. I may check it if you push temporary build into your docker registry and provide me tag of this image.

**Comment by valyala** [comment]
@dmitry-tiger , the bugfix is available in [v1.35.0](https://github.com/VictoriaMetrics/VictoriaMetrics/releases). Closing the issue as fixed. Feel free re-opening it if the issue can be reproduced in `v1.35.0` or newer releases.
